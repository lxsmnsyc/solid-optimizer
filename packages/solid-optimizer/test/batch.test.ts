import { describe, expect, it } from 'vitest';
import compile from './compile';

describe('batch', () => {
  describe('callbacks w/o body', () => {
    it('should work for ImportSpecifier', async () => {
      expect(await compile(`
        import { batch } from 'solid-js';
        
        batch(() => update());
      `)).toMatchSnapshot();
    });
    it('should work for aliased ImportSpecifier', async () => {
      expect(await compile(`
        import { batch as x } from 'solid-js';
        
        x(() => update());
      `)).toMatchSnapshot();
    });
    it('should work for aliased string ImportSpecifier', async () => {
      expect(await compile(`
        import { 'batch' as x } from 'solid-js';
        
        x(() => update());
      `)).toMatchSnapshot();
    });
    it('should work for namespace', async () => {
      expect(await compile(`
        import * as solid from 'solid-js';
        
        solid.batch(() => update());
      `)).toMatchSnapshot();
    });
  });
  describe('callbacks with body', () => {
    it('should work for ImportSpecifier', async () => {
      expect(await compile(`
        import { batch } from 'solid-js';
        
        batch(() => {
          update();
        });
      `)).toMatchSnapshot();
    });
    it('should work for aliased ImportSpecifier', async () => {
      expect(await compile(`
        import { batch as x } from 'solid-js';
        
        x(() => {
          update();
        });
      `)).toMatchSnapshot();
    });
    it('should work for aliased string ImportSpecifier', async () => {
      expect(await compile(`
        import { 'batch' as x } from 'solid-js';
        
        x(() => {
          update();
        });
      `)).toMatchSnapshot();
    });
    it('should work for namespace', async () => {
      expect(await compile(`
        import * as solid from 'solid-js';
        
        solid.batch(() => {
          update();
        });
      `)).toMatchSnapshot();
    });
  });
  describe('spreads', () => {
    it('should work for ImportSpecifier', async () => {
      expect(await compile(`
        import { batch } from 'solid-js';
        
        batch(...example);
      `)).toMatchSnapshot();
    });
    it('should work for aliased ImportSpecifier', async () => {
      expect(await compile(`
        import { batch as x } from 'solid-js';
        
        x(...example);
      `)).toMatchSnapshot();
    });
    it('should work for aliased string ImportSpecifier', async () => {
      expect(await compile(`
        import { 'batch' as x } from 'solid-js';
        
        x(...example);
      `)).toMatchSnapshot();
    });
    it('should work for namespace', async () => {
      expect(await compile(`
        import * as solid from 'solid-js';
        
        solid.batch(...example);
      `)).toMatchSnapshot();
    });
  });
});
