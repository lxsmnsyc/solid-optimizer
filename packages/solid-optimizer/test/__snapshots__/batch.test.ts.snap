// Vitest Snapshot v1

exports[`batch > callbacks w/o body > should work for ImportSpecifier 1`] = `
"import { batch } from 'solid-js';
update();"
`;

exports[`batch > callbacks w/o body > should work for aliased ImportSpecifier 1`] = `
"import { batch as x } from 'solid-js';
update();"
`;

exports[`batch > callbacks w/o body > should work for aliased string ImportSpecifier 1`] = `
"import { 'batch' as x } from 'solid-js';
update();"
`;

exports[`batch > callbacks w/o body > should work for namespace 1`] = `
"import * as solid from 'solid-js';
update();"
`;

exports[`batch > callbacks with body > should work for ImportSpecifier 1`] = `
"import { batch } from 'solid-js';
(() => {
  update();
})();"
`;

exports[`batch > callbacks with body > should work for aliased ImportSpecifier 1`] = `
"import { batch as x } from 'solid-js';
(() => {
  update();
})();"
`;

exports[`batch > callbacks with body > should work for aliased string ImportSpecifier 1`] = `
"import { 'batch' as x } from 'solid-js';
(() => {
  update();
})();"
`;

exports[`batch > callbacks with body > should work for namespace 1`] = `
"import * as solid from 'solid-js';
(() => {
  update();
})();"
`;

exports[`batch > spreads > should work for ImportSpecifier 1`] = `
"import { batch } from 'solid-js';
example[0]();"
`;

exports[`batch > spreads > should work for aliased ImportSpecifier 1`] = `
"import { batch as x } from 'solid-js';
example[0]();"
`;

exports[`batch > spreads > should work for aliased string ImportSpecifier 1`] = `
"import { 'batch' as x } from 'solid-js';
example[0]();"
`;

exports[`batch > spreads > should work for namespace 1`] = `
"import * as solid from 'solid-js';
example[0]();"
`;
